pkgs/by-name/ni/nix-required-mounts/main.py

```diff
@@ -0,0 +1,121 @@
+import glob
+import json
+import subprocess
+import textwrap
+from argparse import ArgumentParser
+from itertools import chain
+from pathlib import Path
+from sys import stderr
+from typing import List
+
+CONFIG = {
+    "nixExe": "nix",
+    "allowedPatterns": {},
+}
+
+parser = ArgumentParser("pre-build-hook")
+parser.add_argument("derivation_path")
+parser.add_argument("sandbox_path", nargs="?")
+parser.add_argument(
+    "--issue-command",
+    choices=("always", "conditional", "never"),
+    default="conditional",
+    help="Whether to print extra-sandbox-paths",
+)
+parser.add_argument(
+    "--issue-stop",
+    choices=("always", "conditional", "never"),
+    default="conditional",
+    help="Whether to print the final empty line",
+)
+
+
+def symlink_parents(p: Path) -> List[Path]:
+    out = []
+    while p.is_symlink() and p not in out:
+        p = p.readlink()
+        out.append(p)
+    return out
+
+
+def entrypoint():
+    if __name__ != "__main__":
+        return
```

Comment:
this pattern is unconventional... why not
```
def entrypoint():
    ...

if __name__ == "__main__":
    entrypoint()
```

Reply:
Because I wanted this to be called _after_ the redefinition of CONFIG 🤡

Reply:
resolving this and switching convo to https://github.com/NixOS/nixpkgs/pull/256230/files#r1362949109